### PR TITLE
server: correctly set hostname

### LINKED ISSUE #898: Hostname not set when using hostNetwork
Hi,

The container's hostname are not being set when using  `hostNetwork:true`.

Here is a sample of a pod in a `CRI-O` node:

```shell
 kubectl exec -it snap-xmfl3 -n kube-system  sh
# echo $HOSTNAME

# hostname
mrsdalloway
# cat /etc/hostname
vmksrh7
```

Another pod from the same `daemonset` in a Docker node:
```shell
kubectl exec -it snap-vfj8r -n kube-system sh
# echo $HOSTNAME
k8s-minion-be-1.rjocta012ahobe-162.cp
# hostname
k8s-minion-be-1.rjocta012ahobe-162.cp
# cat /etc/hostname
k8s-minion-be-1.rjocta012ahobe-162.cp
````

The container's hostname should be set to node's hostname.

### LINKED ISSUE #911: kube-dns selinux AVCs
When running `hack/local-up-cluster.sh` during kubernetes development, kube-dns is deployed by default but fails to run due to blocked access on `/etc/resolv.conf`.  Also ran `ping` which needs ioctl access on that file (why?).  On F26:

```
AVC avc:  denied  { read } for  pid=11278 comm="kube-dns" name="resolv.conf" dev="tmpfs" ino=229066 scontext=system_u:system_r:container_t:s0:c422,c590 tcontext=system_u:object_r:container_var_run_t:s0 tclass=file permissive=0
AVC avc:  denied  { open } for  pid=15813 comm="kube-dns" path="/etc/resolv.conf" dev="tmpfs" ino=254881 scontext=system_u:system_r:container_t:s0:c160,c359 tcontext=system_u:object_r:container_var_run_t:s0 tclass=file permissive=0
AVC avc:  denied  { ioctl } for  pid=21722 comm="ping" path="/etc/resolv.conf" dev="tmpfs" ino=283126 ioctlcmd=0x5401 scontext=system_u:system_r:container_t:s0:c225,c880 tcontext=system_u:object_r:container_var_run_t:s0 tclass=file permissive=0
```

The following TE policy addresses this

```
module kubedns 1.0;

require {
        type container_var_run_t;
        type container_t;
        class file { open read ioctl };
}

#============= container_t ==============
allow container_t container_var_run_t:file read;
allow container_t container_var_run_t:file open;
allow container_t container_var_run_t:file ioctl;
```
@mrunalp @rhatdan @derekwaynecarr

### PR DESCRIPTION
**Pull request by runcom**
Fix #898 
Fix #911 

@mrunalp @sameo PTAL (basically replicated what docker does as well)

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
Marking it WIP as I'm adding some tests. But it fixes the issue reported:
```sh
11:12:54 [src/k8s.io/kubernetes] ‹runtime-stats*› k exec -ti lifecycle-demo-5 sh               / # cat /etc/hostname
runcom.usersys.redhat.com
/ # hostname
runcom.usersys.redhat.com
/ # echo $HOSTNAME
runcom.usersys.redhat.com
```

**Comment by runcom** [comment]
[retest]

File: pkg/annotations/annotations.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ const (
 	// ResolvPath is the resolver configuration path annotation
 	ResolvPath = "io.kubernetes.cri-o.ResolvPath"
 
+	// HostnamePath is the path to /etc/hostname to bind mount annotation
```
Comment (TomSweeneyRedHat) [inline pkg/annotations/annotations.go]: nit:  Can you move this to line ~21 to keep this alphabetical?

**Comment by runcom** [comment]
[retest]

**Comment by mrunalp** [comment]
Restarted travis.

**Comment by mrunalp** [comment]
lint failures.

**Comment by runcom** [comment]
OMG really... lol I'll fix em

**Comment by runcom** [comment]
[retest]

**Comment by mrunalp** [comment]
[retest]

**Comment by mrunalp** [comment]
@cevich CI is failing to run on this. This is blocking release of rc2. Could you please take a look?

**Comment by sameo** [comment]
LGTM

**Comment by rhatdan** [comment]
LGTM

**Pull request closed without merging**